PyInstaller/archive/writers.py

```diff
@@ -363,14 +363,10 @@ def add(self, entry):
                 header = fh.read(4)
                 fh.seek(0)
                 if header == BYTECODE_MAGIC:
-                    # Read whole header and load code. According to PEP-552, in python versions prior to 3.7, the PYC
-                    # header consists of three 32-bit words (magic, timestamp, and source file size).
-                    # From python 3.7 on, the PYC header was extended to four 32-bit words (magic, flags, and, depending
-                    # on the flags, either timestamp and source file size, or a 64-bit hash).
-                    if is_py37:
-                        header = fh.read(16)
-                    else:
-                        header = fh.read(12)
+                    # Read whole header and load code. According to PEP-552, the PYC header consists of four 32-bit
+                    # words (magic, flags, and, depending on the flags, either timestamp and source file size, or a
+                    # 64-bit hash).
+                    header = fh.read(12)
```

Comment:
We need to retain the python 3.7 branch here, not the else one. (I.e., the header is now always 16 bytes).